### PR TITLE
fix: update golden due to STA changes

### PR DESCRIPTION
**Pull request by openroadie**
draft PR (need a PR to OpenRoad/IOpenSTA to be merged to make this PR work).

Fixes the hang in asap7/swerve_* + other fixes.

### CODE REVIEW

**Comment by maliberty** [comment]
I only see updates to the psm tests and not update to the sta submodule.  unit test failures.

**Comment by openroadie** [comment]
@maliberty I could not update the submodule since it was not merged. This is why this PR was in draft status. Will fix.

**Comment by openroadie** [comment]
@maliberty There is a regression failure in STA. I need to update the golden (it is the same as all the psm failures) and then update STA again. I am fine if we want to merge this PR and then do another PR just for 1 ok file depending on urgency

**Comment by openroadie** [comment]
@maliberty I don't quite understand the conflict. Should we just use the new output (Differences found)? Other than this, the PR is ready to go. 

<<<<<<< master
Differences found at line 2.
buffer41, 68.07, 89.6, 1.09978
buffer41, 68.07, 89.6, 1.09985
=======
No differences found.
No differences found.
>>>>>>> master

**Comment by maliberty** [comment]
You need to resolve the conflict in git and push the resolution.  This can't be merged as is.

File: src/dbSta/src/dbSdcNetwork.cc [hunk 1]
```diff
@@ -240,13 +242,13 @@ escapeDividers(const char* token, const Network* network)
...
       '\0',
-      network->pathEscape());
+      network->pathEscape()).c_str();
 }
 
 static const char*
 escapeBrackets(const char* token, const Network* network)
 {
-  return escapeChars(token, '[', ']', network->pathEscape());
+  return escapeChars(token, '[', ']', network->pathEscape()).c_str();
```
[the remark this answers is not in the document]
Comment (openroadie) [reply src/dbSta/src/dbSdcNetwork.cc:251]: @maliberty I pulled the latest OpenSTA code and it has these API changes.  First cut, I just wanted to change the code enough to compile but agree that these suggestions are correct. What would be the best way to fix them? Change the function signature to be a string and then have all the callers deal with it as well? Hopefully that is not too many places.

[the remark this answers is not in the document]
Comment (maliberty) [reply src/dbSta/src/dbSdcNetwork.cc:251]: Yes I would change the return type.

File: src/dbSta/src/dbSdcNetwork.cc [hunk 2]
```diff
@@ -240,13 +242,13 @@
...
```
[the remark this answers is not in the document]
Comment (openroadie) [reply src/dbSta/src/dbSdcNetwork.cc]: Fixed

File: src/dbSta/src/dbSdcNetwork.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (openroadie) [reply src/dbSta/src/dbSdcNetwork.cc]: Fixed.

File: src/dst/CMakeLists.txt [hunk 3]
```diff
@@ -62,6 +64,8 @@ target_link_libraries(dst
   PUBLIC
     utl
     OpenSTA
+    tcl
+    ${ZLIB_LIBRARIES}
```
Comment (maliberty) [inline src/dst/CMakeLists.txt:68]: Is this going to be fixed in the next sta update?

**Review by maliberty — APPROVED** [review]

File: src/dst/CMakeLists.txt (hunk 3 above)
Comment (openroadie) [reply src/dst/CMakeLists.txt:68]: I think this is already fixed but let me verify that before I undo this change.

**Pull request merged**